source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c

```diff
@@ -304,6 +333,85 @@ static BaseType_t xSTM32H_NetworkInterfaceInitialise( NetworkInterface_t * pxInt
             HAL_ETH_DescAssignMemory( &( xEthHandle ), uxIndex, pucBuffer, NULL );
         }
 
+        #if ( ipconfigUSE_MDNS == 1 )
+            {
+                /* Program the MDNS address. */
+                prvMACAddressConfig( &xEthHandle, xMACEntry, ( uint8_t * ) xMDNS_MacAdress.ucBytes );
+                xMACEntry += 8;
+            }
+        #endif
+        #if ( ( ipconfigUSE_MDNS == 1 ) && ( ipconfigUSE_IPv6 != 0 ) )
+            {
+                prvMACAddressConfig( &xEthHandle, xMACEntry, ( uint8_t * ) xMDNS_MACAdressIPv6.ucBytes );
+                xMACEntry += 8;
+            }
+        #endif
+        #if ( ipconfigUSE_LLMNR == 1 )
+            {
+                /* Program the LLMNR address. */
+                prvMACAddressConfig( &xEthHandle, xMACEntry, ( uint8_t * ) xLLMNR_MacAdress.ucBytes );
+                xMACEntry += 8;
+            }
+        #endif
+        #if ( ( ipconfigUSE_LLMNR == 1 ) && ( ipconfigUSE_IPv6 != 0 ) )
+            {
+                prvMACAddressConfig( &xEthHandle, xMACEntry, ( uint8_t * ) xLLMNR_MacAdressIPv6.ucBytes );
+                xMACEntry += 8;
+            }
+        #endif
+
+        {
+            /* The EMAC address of the first end-point has been registered in HAL_ETH_Init(). */
+            for( ;
+                 pxEndPoint != NULL;
+                 pxEndPoint = FreeRTOS_NextEndPoint( pxMyInterface, pxEndPoint ) )
+            {
+                #if ( ipconfigUSE_IPv6 != 0 )
+                    if( pxEndPoint->bits.bIPv6 != pdFALSE_UNSIGNED )
+                    {
+                        uint8_t ucMACAddress[ 6 ] = { 0x33, 0x33, 0xff, 0, 0, 0 };
+
+                        ucMACAddress[ 3 ] = pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 13 ];
+                        ucMACAddress[ 4 ] = pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 14 ];
+                        ucMACAddress[ 5 ] = pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 15 ];
+
+                        /* Allow traffic destined to Solicited-Node multicast address of this endpoint
+                         * for Duplicate Address Detection (DAD) */
+                        prvMACAddressConfig( &xEthHandle, xMACEntry, ucMACAddress );
+                        xMACEntry += 8;
+                    }
+                    else
```

Comment:
The else (line 383) here is combined with line 394. ( `if( xMACEntry > ( BaseType_t ) ETH_MAC_ADDRESS3 )` )

I think you're using else for IPv4. So please correct the logic here.

Reply:
Updated